crates/pep440-rs/src/version.rs

```diff
@@ -2320,7 +2412,13 @@ pub(crate) fn compare_release(this: &[u64], other: &[u64]) -> Ordering {
 ///
 /// [pep440-suffix-ordering]: https://peps.python.org/pep-0440/#summary-of-permitted-suffixes-and-relative-ordering
 fn sortable_tuple(version: &Version) -> (u64, u64, Option<u64>, u64, &[LocalSegment]) {
-    match (version.pre(), version.post(), version.dev(), version.min()) {
+    // If the version is a "max" version, use a post version larger than any possible post version.
+    let post = if version.is_max() {
+        Some(u64::MAX)
+    } else {
+        version.post()
+    };
```

Comment:
We could also consider incorporating `max` into the `match` below... This felt easier to get right.